src/components/sbb-datepicker/sbb-datepicker.tsx

```diff
@@ -238,8 +218,12 @@ export class SbbDatepicker implements ComponentInterface {
   /** Applies the correct format to values and triggers event dispatch. */
   private async _formatAndUpdateValue(value: string): Promise<void> {
     if (this._inputElement) {
-      this._inputElement.value = this._formatValue(value);
-      const newValueAsDate: Date = await this.getValueAsDate();
+      const newValueAsDate: Date = this._parse(value);
+
+      this._inputElement.value = this._dateAdapter.isValid(newValueAsDate)
```

Comment:
This seems very similar to the _formatValue method. Should it maybe be split so part of it could be used here?